src/pivot-table/hooks/use-selections-model.ts

```diff
@@ -48,68 +86,68 @@ export default function useSelectionsModel(
   }, [selections, updatePageInfo]);
 
   const isLocked = useCallback(
-    (qType: EngineAPI.NxSelectionCellType, qRow: number, qCol: number) => {
-      switch (qType) {
-        case NxSelectionCellType.NX_CELL_LEFT:
-          return !!selected.find(
-            (cell) =>
-              cell.qType === NxSelectionCellType.NX_CELL_TOP ||
-              (cell.qType === NxSelectionCellType.NX_CELL_LEFT && cell.qCol !== qCol),
-          );
-        case NxSelectionCellType.NX_CELL_TOP:
-          return !!selected.find(
-            (cell) =>
-              cell.qType === NxSelectionCellType.NX_CELL_LEFT ||
-              (cell.qType === NxSelectionCellType.NX_CELL_TOP && cell.qRow !== qRow),
-          );
-        default:
-          return false;
+    (cell: Cell) => {
+      if (selectedPivotCells.size === 0 || selectedRowOrColumn.selectionCellType === null) {
+        return false;
       }
+
+      if (cell.selectionCellType !== selectedRowOrColumn.selectionCellType) {
+        return true;
+      }
+
+      if (cell.selectionCellType === NxSelectionCellType.NX_CELL_LEFT) {
+        return selectedRowOrColumn.coord !== cell.x;
+      }
+
+      return selectedRowOrColumn.coord !== cell.y;
     },
-    [selected],
+    [selectedPivotCells, selectedRowOrColumn],
   );
 
   const select = useCallback(
-    (qType: EngineAPI.NxSelectionCellType, qRow: number, qCol: number) => async (evt: React.MouseEvent) => {
-      if ((evt.target as HTMLElement)?.className.includes("sn-pivot-table-column-adjuster")) {
+    (cell: Cell) => async (evt: React.MouseEvent) => {
+      if ((evt.target as HTMLElement | SVGElement)?.getAttribute("class") === "sn-pivot-table-column-adjuster") {
```

Comment:
When clicking on the expand/collapse icon, you get an SVG element. Which does not behave the same when `className` is called (https://developer.mozilla.org/en-US/docs/Web/API/Element/className#notes)